Chatto/sources/Keyboard/KeyboardUpdatesHandler.swift

```diff
@@ -0,0 +1,170 @@
+//
+// Copyright (c) Badoo Trading Limited, 2010-present. All rights reserved.
+//
+import UIKit
+
+public protocol KeyboardInputAdjustableViewController: UIViewController {
+    var inputBarContainer: UIView { get }
+    var inputContainerBottomConstraint: NSLayoutConstraint { get }
+}
+
+public protocol KeyboardUpdatesHandlerProtocol {
+    var keyboardTrackingView: KeyboardTrackingView { get }
+
+    func adjustLayoutIfNeeded()
+    func startTracking()
+    func stopTracking()
+}
+
+public protocol KeyboardUpdatesHandlerDelegate: AnyObject {
+    func keyboardUpdatesHandler(_: KeyboardUpdatesHandler, didAdjustBottomMarginTo bottomMargin: CGFloat)
```

Comment:
I think we're declaring concrete implementation of `KeyboardUpdatesHandler` here.
I'd remove this parameter completely add rename it to:
`func didAdjustBottomMargin(to margin: CGFloat)`